core/io/darwin.rs

```diff
@@ -24,8 +18,8 @@ impl DarwinIO {
     pub fn new() -> Result<Self> {
         Ok(Self {
             poller: Rc::new(RefCell::new(Poller::new()?)),
-            events: Rc::new(RefCell::new(Events::new())),
-            callbacks: Rc::new(RefCell::new(HashMap::new())),
+            events: <_>::default(),
+            callbacks: <_>::default(),
```

Comment:
why? I don't like this, now it isn't explicit and I have to understand what default does here.

Reply:
Yeah, I tend to agree that explicit is better here, but I think using `default()` is more idiomatic Rust.

Reply:
I hate idiomatic rust then. I have to remember the defaults for everything now? I now what should be the default when I explicitly type them. Idk maybe I'm just a rust hater.

Reply:
Yeah, I hate some idiomatic Rust too, just wanted to give an explanation why someone would change the code like that. We may eventually need to write a Limbo style guide to codify our preferences if we diverge from idiomatic Rust a lot.